mmhuman3d/core/visualization/renderer/torch3d_renderer/pointcloud_renderer.py

```diff
@@ -70,54 +68,53 @@ def __init__(self,
         Returns:
             None
         """
-        self.device = device
-        self.output_path = output_path
-        self.resolution = resolution
-        self.return_type = return_type
-        self.out_img_format = out_img_format
         self.radius = radius
-        self.projection = projection
-        self.in_ndc = in_ndc
-        if output_path is not None:
-            if check_path_suffix(output_path, ['.mp4', '.gif']):
-                self.temp_path = osp.join(
-                    Path(output_path).parent,
-                    Path(output_path).name + '_output_temp')
-                mmcv.mkdir_or_exist(self.temp_path)
-                print('make dir', self.temp_path)
-            else:
-                self.temp_path = output_path
-        self.set_render_params(**kwargs)
-        super(MeshBaseRenderer, self).__init__()
+        super().__init__(
+            resolution=resolution,
+            device=device,
+            output_path=output_path,
+            obj_path=None,
+            return_type=return_type,
+            out_img_format=out_img_format,
+            projection=projection,
+            in_ndc=in_ndc,
+            **kwargs)
+
+    def to(self, device):
+        if isinstance(device, str):
+            device = torch.device(device)
+        self.device = device
+        if self.rasterizer.cameras is not None:
+            self.rasterizer.cameras = self.rasterizer.cameras.to(device)
+        self.compositor = self.compositor.to(device)
+        return self
 
-    def set_render_params(self, **kwargs):
+    def _init_renderer(self, rasterizer, compositor, **kwargs):
```

Comment:
`_init_renderer` also has many duplications in various renderers. Can we also implement it in MeshBaseRenderer?

Reply:
> `_init_renderer` also has many duplications in various renderers. Can we also implement it in MeshBaseRenderer?


Just add default `shader` for different renders if no 'shader' passed. What do you suggest?

Reply:
> > `_init_renderer` also has many duplications in various renderers. Can we also implement it in MeshBaseRenderer?
> 
> Just add default `shader` for different renders if no 'shader' passed. What do you suggest?

I changed this